### PR TITLE
sql: add comment-syntax.md

### PR DESCRIPTION
**Pull request by winkyao**

### CODE REVIEW

File: sql/comment-syntax.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+```
+SELECT /*+ hint */ FROM ...;
+```
+
+由于 hint 包含在类似 /*+ xxx */ 的 comment 里，MySQL 客户端在 5.7.7 之前，会默认把 comment 清除掉，如果需要在旧的客户端使用 hint，需要在启动客户端时加上 --comments 选项，例如 mysql -h 127.0.0.1 -P 4000 -uroot --comments
+
+目前支持的 TiDB 特有的 Hint 有以下：
+
+##### TIDB_SMJ(t1, t2)
```
Comment (andelf) [inline sql/comment-syntax.md]: ..... are you sure, level 5 heading?

**Comment by shenli** [comment]
这篇文档没有链接？从哪里能点进来？

File: sql/comment-syntax.md (hunk 1 above)
Comment (shenli) [reply sql/comment-syntax.md]: 用 * 就差不多了

**Comment by winkyao** [comment]
@shenli https://dev.mysql.com/doc/refman/5.7/en/comments.html

**Comment by shenli** [comment]
LGTM @andelf PTAL

**Pull request merged**